ambry-utils/src/main/java/com.github.ambry.utils/Utils.java

```diff
@@ -682,4 +680,17 @@ public static ByteBuffer ensureCapacity(ByteBuffer existingBuffer, int newLength
     }
     return existingBuffer;
   }
+
+  /**
+   * Gets the root cause for {@code t}.
+   * @param t the {@link Throwable} whose root cause is required.
+   * @return the root cause for {@code t}.
+   */
+  public static Throwable getRootCause(Throwable t) {
```

Comment:
Why does this have to take in/return a `Throwable` instead of an `Exception`? It seems like we just cast the return value into an exception in quite a few cases.